components/common/Header.tsx

```diff
@@ -1,19 +1,104 @@
 import React from "react"
-import { Flex } from "@chakra-ui/react"
-
-function Header(): JSX.Element {
-    return (
-        <Flex
-            paddingTop="1rem"
-            borderBottom="1px"
-            borderColor="gray.400"
-            paddingBottom="1rem"
-            textAlign="center"
-            justifyContent="center"
-            paddingRight="1rem"
-            backgroundColor="white"
-        />
-    )
+import {
+    ChakraProvider,
+    extendTheme,
+    Heading,
+    StyleConfig,
+} from "@chakra-ui/react"
+
+const components: Record<string, StyleConfig> = {
+    Heading: {
+        baseStyle: {
+            paddingX: "2rem",
+            paddingY: "1rem",
+            color: "whitesmoke",
+        },
+        variants: {
+            topHeader: {
+                paddingY: "1.5rem",
+                color: "black",
+                bg: "whitesmoke",
+            },
+            mainHeader: {
+                bg: "black",
+            },
+            subHeader: {
+                bg: "#6f6f6f",
+            },
+            elementHeader: {
+                bg: "#6f6f6f",
+                paddingX: "1.5rem",
+            },
+            tableHeader: {
+                bg: "#6f6f6f",
+                paddingX: "1.5rem",
+            },
+        },
+        defaultProps: {
+            size: "md",
+        },
+    },
 }
 
+const StyledItems = ({ children }: { children: JSX.Element }) => (
+    <ChakraProvider theme={extendTheme({ components })}>
+        {children}
+    </ChakraProvider>
+)
+
+const Header = (): JSX.Element => (
+    <StyledItems>
+        <Heading as="h1" variant="topHeader" size="3xl">
+            {"[...] - Epoc"}
+        </Heading>
+    </StyledItems>
+)
+
+export const MainHeader = ({ text }: { text: string }) => (
+    <StyledItems>
+        <Heading as="h2" variant="mainHeader" size="xl">
+            {text}
+        </Heading>
+    </StyledItems>
+)
+
+export const SubHeader = ({ text }: { text: string }) => (
+    <StyledItems>
+        <Heading as="h3" variant="subHeader" size="md">
+            {text}
+        </Heading>
+    </StyledItems>
```

Comment:
These headers also contain a lot of repetitive code which could be extracted to one function taking the `as`, `variant`, `size` and the contents as parameters